plugins/feeds/public/ipspamlist.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from datetime import timedelta
+
+from core import Feed
+from core.errors import ObservableValidationError
+from core.observables import Ip
+
+
+class IPSpamList(Feed):
+    default_values = {
+        "frequency":
+            timedelta(days=1),
+        "name":
+            "IPSpamList",
+        "source":
+            "http://www.ipspamlist.com/public_feeds.csv",
+        "description":
+            "Service provided by NoVirusThanks that keeps track of malicious "
+            "IP addresses engaged in hacking attempts, spam comments"
+    }
+
+    def update(self):
+        for line in self.update_csv(delimiter=',', quotechar=None):
+            self.analyze(line)
+
+    def analyze(self, item):
+        if not item or item[0].startswith('first_seen'):
+            return
+        try:
+            context = dict([('source', self.name)])
+
+            first_seen, last_seen, ip_address, category, attacks_count = item
+
+            ip = Ip.get_or_create(value=ip_address)
+
+            context['threat'] = category
+            ip.tag(category)
+
+            context['first_seen'] = first_seen
+
+            context['last_seen'] = last_seen
+
+            context['attack_count'] = attacks_count
+
+            ip.add_source('feed')
+
+            ip.add_context(context)
+
+        except ObservableValidationError as e:
```

Comment:
ObservableValidationError cannot happen after line 34, so please move the rest of the code out of the exception.

Also why so many blank lines? I think the code would be just as clear if you removed some of them (e.g. L40, L42, L46...)